data/org.freedesktop.portal.DynamicLauncher.xml

```diff
@@ -0,0 +1,223 @@
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Matthew Leeds <mwleeds@protonmail.com>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.DynamicLauncher:
+      @short_description: Portal for installing application launchers onto the
+        desktop
+
+      The DynamicLauncher portal allows sandboxed applications to install launchers
+      which have an icon associated with them and which execute a command in the
+      sandboxed application. The desktop environment would display the launcher
+      to the user in its menu of installed applications. For example this can be
+      used by a sandboxed browser to install web app launchers.
+
+      When initiated by a sandboxed application, the PrepareInstall() method would
+      first be called and would result in a dialog being presented to the user
+      so they can confirm they want to install the application. Then, the token
+      returned by PrepareInstall() would be passed to the Install() method to
+      install the launcher without user interaction.
+
+      By contrast, when a trusted system component initiates the process, the
+      RequestInstallToken() method is used to acquire a token without showing a
+      dialog to the user. The token is then passed to the Install() method to
+      complete the installation (which means putting the .desktop file and icon
+      in appropriate filesystem locations).
+
+      This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.portal.DynamicLauncher">
+    <!--
+        Install:
+        @token: Token proving authorization of the installation
+        @desktop_file_id: The GApplication ID, used to determine the desktop
+          file name
+        @desktop_entry: The text of the Desktop Entry file to be installed, see below
+        @options: Vardict with optional further information
+        @result: 0 to indicate success, a nonzero number to indicate failure
+
+        Installs a .desktop launcher and icon into appropriate directories to
+        allow the desktop environment to find them.
+
+        @token must be a token that was returned by a previous
+        org.freedesktop.portal.DynamicLauncher.PrepareInstall() or
+        org.freedesktop.portal.DynamicLauncher.RequestInstallToken() call.
+        The token is valid for up to five minutes.
```

Comment:
How many times can this token be used during these 5 minutes?

Reply:
once